### PR TITLE
Added anchors to configuration keys and sections

### PR DESCRIPTION
**Pull request by FroMage**
As requested by @dmlloyd and @maxandersen

### CODE REVIEW

File: core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/DocFormatter.java [hunk 1]
```diff
@@ -2,16 +2,66 @@
...
+                .replaceAll("[ŁĽĹĻĿ]", "L")
+                .replaceAll("[ÑŃŇŅŊ]", "N")
+                .replaceAll("[ÒÓÔÕÖØŌŐŎ]", "O")
+                .replaceAll("[ŔŘŖ]", "R")
+                .replaceAll("[ŚŠŞŜȘ]", "S")
+                .replaceAll("[ÙÚÛÜŪŮŰŬŨŲ]", "U")
+                .replaceAll("[Ŵ]", "W")
+                .replaceAll("[ÝŶŸ]", "Y")
+                .replaceAll("[ŹŽŻ]", "Z")
+                .replaceAll("[ß]", "ss");
```
Comment (dmlloyd) [inline core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/DocFormatter.java]: Very... thorough :)

**Review by dmlloyd — APPROVED** [review]

File: core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/DocFormatter.java (hunk 1 above)
Comment (FroMage) [reply core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/DocFormatter.java]: Yeah, I took this from a French source with years of experience dealing with I18n ;)

File: core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/SummaryTableDocFormatter.java [hunk 2]
```diff
@@ -8,28 +8,35 @@
...
+
     /**
      * Generate configuration keys in table format.
      * Generated table will contain a key column that points to the long descriptive format.
      *
      * @param configDocItems
      */
     @Override
-    public void format(Writer writer, List<ConfigDocItem> configDocItems) throws IOException {
+    public void format(Writer writer, boolean useAnchorPrefixes, List<ConfigDocItem> configDocItems) throws IOException {
```
Comment (machi1990) [inline core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/SummaryTableDocFormatter.java]: Is there any reason we can't generate anchors all the time?

**Comment by geoand** [comment]
CI needs some love :)

**Review by machi1990 — APPROVED** [review]

**Comment by machi1990** [comment]
> CI needs some love :)

Might be related to the fact that 
`agroal`, `reactive-mysql-client`, and `reactive-pg-client` share the same config keys `datasource.url`, `datasource.username` ...

File: core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/SummaryTableDocFormatter.java (hunk 2 above)
Comment (FroMage) [reply core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/SummaryTableDocFormatter.java]: I think we do, but they don't need a prefix if we're generating a single page.

**Comment by FroMage** [comment]
Ah, I didn't think of the fact that one doc file can include two config files with the same keys. Lemme find a fix.

**Comment by FroMage** [comment]
Hopefully fixed, and rebased on master. Let's wait for CI.

**Pull request merged**